lua/clangd_extensions/type_hierarchy.lua

```diff
@@ -5,10 +5,17 @@ local nvim_get_current_buf = api.nvim_get_current_buf
 local type_hierarchy_augroup =
     api.nvim_create_augroup("ClangdTypeHierarchy", {})
 
+---@class ClangdTypeHierarchy
 local M = {}
 M.type_to_location = {}
 M.offset_encoding = {}
 
+---@param node any
+---@param visited table|unknown
+---@param result table|unknown
+---@param padding string
+---@param type_to_location unknown
```

Comment:
`type_to_location` is a table, `node` is a table too iirc - we can review all `node`s in the end